### ISSUE TITLE
UHK not detected by Agent 2.0.0 on MacBook M1

### ISSUE THREAD

**Issue by WindChimeRan** [UltimateHackingKeyboard/agent#1873]
I'm using UHKv1 + m1pro mbp + usb dock (I don't have usb-c cables)

With the newest released agent-v2: "no device detected". (I tried both UHK.Agent-2.0.0-mac.zip and UHK.Agent-2.0.0-mac.dmg)

Building the electron application locally: works perfectly.

**Comment by gerbrand-bosch** [comment]
It seems like the UHK.Agent-2.0.0 release is build for the x86_64 architecture instead of the arm64e needed for the M1/M2 chip.
```
[DeviceService] Device connection state query error Error: dlopen(/var/folders/zh/4_01swws19g1gnmh7nv97xh00000gn/T/.com.ultimategadgetlabs.agent.yYn5js, 0x0001): 
tried: '/var/folders/zh/4_01swws19g1gnmh7nv97xh00000gn/T/.com.ultimategadgetlabs.agent.yYn5js' 
(mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64e')), '/private/var/folders/zh/4_01swws19g1gnmh7nv97xh00000gn/T/.com.ultimategadgetlabs.agent.yYn5js' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64e'))
    at process.func [as dlopen] (node:electron/js2c/asar_bundle:5:1812)
    at Module._extensions..node (node:internal/modules/cjs/loader:1205:18)
    at Object.func [as .node] (node:electron/js2c/asar_bundle:5:2039)
    at Module.load (node:internal/modules/cjs/loader:988:32)
    at Module._load (node:internal/modules/cjs/loader:829:12)
    at c._load (node:electron/js2c/asar_bundle:5:13343)
    at Module.require (node:internal/modules/cjs/loader:1012:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at bindings (/Applications/UHK Agent.app/Contents/Resources/app.asar/node_modules/bindings/bindings.js:112:48)
    at loadBinding (/Applications/UHK Agent.app/Contents/Resources/app.asar/node_modules/node-hid/nodehid.js:25:42)
```

**Comment by gerbrand-bosch** [comment]
This applies to the download of the github release and the upgrade process using the older 1.5.17 agent.

**Comment by mondalaci** [comment]
@ert78gb Can you look into this?

**Comment by ert78gb** [comment]
We create the releases on GitHub CI and it currently does not support Mx chips. 

I built the following 2 files on my machine and uploaded to [google drive](https://drive.google.com/drive/u/0/folders/1Sxy45EFbLmbdM41KknTKdBdpplAZ5NOD) because GitHub allows to attach files that size less then 25 MB

Could you check it works on your machine. 
I will also test them on non Mx architecture

**Comment by gerbrand-bosch** [comment]
Yes, the agent now asks to update the firmware as expected.
![Screenshot 2022-10-24 at 10 53 36](https://user-images.githubusercontent.com/12494166/197487548-73070a1c-75e7-46bc-97b1-8c6d54e57b8b.png)

**Comment by danpalmer** [comment]
FYI, a temporary workaround for this is to force running the whole app under x86:

```
$ arch -x86_64 /Applications/UHK\ Agent.app/Contents/MacOS/UHK\ Agent
```

**Comment by spuder** [comment]
Even though Github CI does not yet offer ARM64 runners, it is still possible to build ARM64 compatible electron apps on x86 hardware

See this documentation https://www.electronjs.org/blog/apple-silicon

**Comment by WindChimeRan** [comment]
> We create the releases on GitHub CI and it currently does not support Mx chips.
> 
> I built the following 2 files on my machine and uploaded to [google drive](https://drive.google.com/drive/u/0/folders/1Sxy45EFbLmbdM41KknTKdBdpplAZ5NOD) because GitHub allows to attach files that size less then 25 MB
> 
> Could you check it works on your machine. I will also test them on non Mx architecture

Yes it works with my m1 pro machine. Thanks!

**Comment by TrevorAC99** [comment]
I also experience this on an Intel Windows system with Agent 2.0.0 and a brand new UHKv2. It works fine when I run from source (which is what I am doing so that I can configure the keyboard). When I open the debug console, I see the infinite loop of errors mentioned in #1876.

I can provide additional details as needed.

**Comment by ert78gb** [comment]
Hi @TrevorAC99 ,

What is version of your Windows?

**Comment by TrevorAC99** [comment]
@ert78gb 
Edition: Windows 11 Pro
Version: 22H2
OS Build: 22621.674

Just in case this is relevant
CPU: i7-12700K

**Comment by mondalaci** [comment]
The newly released Agent 2.0.1 should solve this issue.
